girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -69,21 +73,84 @@ def create_dandiset(self, params):
     def get_dandiset(self, identifier, params):
-
-        if not identifier:
-            raise RestException("identifier must not be empty.")
-
-        if not validate_dandiset_identifier(identifier):
-            raise RestException("Invalid Dandiset Identifier")
-
         # Ensure we are only looking for drafts collection child folders.
         drafts = get_or_create_drafts_collection()
         doc = Folder().findOne({"parentId": drafts["_id"], "meta.dandiset.identifier": identifier})
         if not doc:
             raise RestException("No such dandiset found.")
         return doc
 
+    @access.public
+    @describeRoute(
+        Description("Get Dandiset Owners").param(
+            "identifier", "Dandiset Identifier", paramType="path"
+        )
+    )
+    @dandiset_identifier
+    def get_dandiset_owners(self, identifier, params):
+        return Folder().getFullAccessList(find_dandiset_by_identifier(identifier))["users"]
+
+    @access.user
+    @autoDescribeRoute(
+        Description("Add Dandiset Owners")
+        .param("identifier", "Dandiset Identifier", paramType="path")
+        .jsonParam(
+            "owners",
+            "A JSON list of girder users to add as owners.",
+            paramType="body",
+            requireArray=True,
+        )
+    )
+    @dandiset_identifier
+    def add_dandiset_owners(self, identifier, owners, params):
+        dandiset = find_dandiset_by_identifier(identifier)
+        Folder().requireAccess(dandiset, user=self.getCurrentUser(), level=AccessType.ADMIN)
+
+        # Make sure the list doesn't contain duplicates
+        current_owners = [
+            (str(user["id"]), user["level"])
+            for user in Folder().getFullAccessList(dandiset)["users"]
+        ]
+        new_owners = [(user["_id"], AccessType.ADMIN) for user in owners]
+        final_owners = [
+            {"id": user_id, "level": level}
+            for user_id, level in set([*current_owners, *new_owners])
+        ]
+
+        doc = Folder().setAccessList(dandiset, {"users": final_owners}, save=True, recurse=True)
+        return Folder().getFullAccessList(doc)
+
+    @access.user
+    @autoDescribeRoute(
+        Description("Remove Dandiset Owners")
+        .param("identifier", "Dandiset Identifier", paramType="path")
+        .jsonParam(
+            "owners",
+            "A JSON list of girder users to remove from owners.",
+            paramType="body",
+            requireArray=True,
+        )
+    )
+    @dandiset_identifier
+    def remove_dandiset_owners(self, identifier, owners, params):
+        dandiset = find_dandiset_by_identifier(identifier)
+        Folder().requireAccess(dandiset, user=self.getCurrentUser(), level=AccessType.ADMIN)
+
+        current_owners = {
+            (str(user["id"]), user["level"])
+            for user in Folder().getFullAccessList(dandiset)["users"]
+        }
+        owners_to_delete = {(user["_id"], AccessType.ADMIN) for user in owners}
```

Comment:
should you do some validation of passed in list of users here and in other endpoints?

Reply:
I had a similar thought, but I'm not sure what I'd filter it on?

Reply:
However if you mean validating that the passed in user objects themselves are valid user objects then I agree, that's something I'm going to add.

Reply:
94b1e48